src/protocols/protocol_address_31402.cpp

```diff
@@ -102,22 +99,17 @@ bool protocol_address_31402::handle_receive_get_address(const code& ec,
     if (stopped(ec))
         return false;
 
-    // TODO: allowing repeated queries can allow a channel to map our history.
-    // TODO: pull active hosts from host cache (currently just resending self).
-    // TODO: need to distort for privacy, don't send currently-connected peers.
-    // TODO: response size limit is max_address (1000).
-
-    if (self_.addresses().empty())
-        return false;
-
-    LOG_DEBUG(LOG_NETWORK)
-        << "Sending addresses to [" << authority() << "] ("
-        << self_.addresses().size() << ")";
+    if (!self_.addresses().empty())
+    {
+        SEND2(self_, handle_send, _1, self_.command);
```

Comment:
See above comments, this should SEND(addresses, ....) where addresses is created locally.

Reply:
Ok

Reply:
This part, I'm not sure I understand.  I see how to either send the self object like it was originally, or creating a new one for sending such as:

```
// where addresses is a local
const address address_subset(addresses);
SEND2(address_subset, handle_send, _1, self_.command);
```

Am I missing a macro for sending the non-class wrapped address::list?

Reply:
You can only `SEND[X](foo,, ...)` where `foo` is a `bc::message` type.